bridge/src/lib.rs

```diff
@@ -302,6 +330,85 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Mark the give dest domainID with chainID to be enabled
+		#[pallet::weight(195_000_000)]
+		pub fn register_domain(
+			origin: OriginFor<T>,
+			dest_domain_id: DomainID,
+			dest_chain_id: ChainID,
+		) -> DispatchResult {
+			let mut sender: T::AccountId = [0u8; 32].into();
+			if <T as Config>::BridgeCommitteeOrigin::ensure_origin(origin.clone()).is_err() {
```

Comment:
hmm.. I'm not sure if I understand you here @tolak. If the origin is `root`, the logic will skip the if condition, if not, the `<T as Config>::BridgeCommitteeOrigin::ensure_origin(origin.clone())` will return an error so that the logic will fall into the if condition to check if the signer has enough permission by `sygma_access_segregator`. if signer doesn't have permission, the error will be `AccessDenied`

Reply:
Oh, I lost my mind, thanks for clarify